### ISSUE TITLE
CI runs every test job in all python environments

### ISSUE THREAD

**Issue by sfmig** [neuroinformatics-unit/movement#260]
**Describe the bug**
From the tests workflow file it seems like we would like to run tests:
- with all supported Python versions, on Linux; and
- with Python 3.11 on `macos-13`, `macos-latest` and `windows-latest`.

However, it seems like we are running the test suite with all Python versions, on all platforms.

Am I missing something obvious? 🤔 

**To Reproduce**
This can be seen in the [test logs](https://github.com/neuroinformatics-unit/movement/actions/workflows/test_and_deploy.yml). For example:

- the job called `ubuntu-latest py3.10` runs tox in a `py310` environment, but also `py311` ([here](https://github.com/neuroinformatics-unit/movement/actions/runs/10253265176/job/28365526876?pr=259#step:4:537)) and `py312`  ([here](https://github.com/neuroinformatics-unit/movement/actions/runs/10253265176/job/28365526876?pr=259#step:4:874))
- same with the `windows-latest py3.11` of the same run: it runs tox with `py310` ([here](https://github.com/neuroinformatics-unit/movement/actions/runs/10253265176/job/28365528503?pr=259#step:4:2400)) and `py312` too ([here](https://github.com/neuroinformatics-unit/movement/actions/runs/10253265176/job/28365528503?pr=259#step:4:3074))

**Expected behaviour**
I would expect that 
- `ubuntu-latest py3.10` runs tox with the `py310` environment only
- `ubuntu-latest py3.11` runs tox with the `py311` environment only
- `ubuntu-latest py3.12` runs tox with the `py312` environment only
- etc

**Log file**
Please see logs linked above.

**Screenshots**
\

**Computer used (please complete the following information):**
This can be checked in the logs of the github runners

**Additional context**
\

**Comment by niksirbi** [comment]
Good spotting @sfmig! We should bring this up to the team meeting (or slack), because I feel like this could be the case for our other packages as well.

**Comment by adamltyson** [comment]
it might just be worth getting rid of tox, I'm not sure it simplifies stuff as much as it complicates it.

**Comment by sfmig** [comment]
> I feel like this could be the case for our other packages as well.

I checked a few `brainglobe` repos and it wasn't the case. 

The issue came about because of the introduction of the `tox requires` section. If defined, we need to explicitly add `tox-gh-actions` plugin to that section for the plugin to work (see [here](https://github.com/ymyzk/tox-gh-actions#tox-requires)). I don't think the `tox requires` section is used [in brainglobe](https://github.com/search?q=org%3Abrainglobe+requires+%3D+tox-&type=code).

It is possible that the alternative plugin `tox-gh` (developed by `tox`) does not have this "issue" - this led me to raise [this issue](https://github.com/neuroinformatics-unit/actions/issues/60). 

> it might just be worth getting rid of tox

@adamltyson I'm curious about this opinion. I actually find `tox` really useful, since it makes testing in different Python versions and environments very systematic. The alternative would be for us to manually define this in CI, which seems more error prone, right? I also find the syntax quite compact.

Maybe if it is a confusing tool we could do a deep dive and present it in a group meeting? I wouldn't mind doing that.

**Comment by niksirbi** [comment]
I see, thanks for the explanation @sfmig !

**Comment by adamltyson** [comment]
>  actually find tox really useful, since it makes testing in different Python versions and environments very systematic. The alternative would be for us to manually define this in CI, which seems more error prone, right? I also find the syntax quite compact.

Don't we do that anyway with CI? We specify Python versions and environments in a GH actions test matrix.

**Comment by sfmig** [comment]
> Don't we do that anyway with CI?

Yes, and with `tox` it is quite easy to map Github actions combinations of Python versions and platform to a `tox` testing environment. My understanding is that if we did this manually instead in Github actions (i.e., without `tox`), it would involve a few explicit calls to create and ensure isolated environments and run pytest.

My impression is that `tox` makes running tests locally and in CI more similar, which often makes it easier to debug problems.

For example, when transitioning `cellfinder` from `tf` to `keras`, we could run tests in an environment with TF only, or with Pytorch only, and with different environment variables set, which was very handy for sanity checks (see [here](https://github.com/brainglobe/cellfinder/blob/89339fabfa25a4c9cd9db155ea4975b21fdeda94/pyproject.toml#L126)). We could do these things via Github actions too, but without tox we lose the option to run something equivalent locally, which imo is an important benefit.

I am no expert in `tox` and have only scratched the surface, just saying I have found it useful in the past.
